coil/src/main/java/dev/chrisbanes/accompanist/coil/Crossfade.kt

```diff
@@ -0,0 +1,263 @@
+ * has been loaded from memory.
+ * @param durationMs The duration of the crossfade animation in milliseconds.
+ * @param clock The animation clock.
+ */
+@Composable
+private fun crossfadePainter(
+    result: SuccessResult,
+    skipFadeWhenLoadedFromMemory: Boolean = true,
+    durationMs: Int = DefaultTransitionDuration,
+    clock: AnimationClockObservable = AnimationClockAmbient.current.asDisposableClock()
+): Painter {
+    return if (skipFadeWhenLoadedFromMemory && result.isFromMemory) {
+        // If can skip the fade when loaded from memory, we do not need to run the animation on it
+        defaultSuccessPainterGetter(result)
+    } else {
+        val observablePainter = remember {
+            ObservableCrossfade(result, durationMs, clock).also { it.start() }
+        }
+        if (!observablePainter.isFinished) {
+            // If the animation is running, using it's painter
+            observablePainter.painter
+        } else {
+            // If the animation has finished, revert back to the default painter
+            defaultSuccessPainterGetter(result)
+        }
+    }
+}
+
+@Suppress("JoinDeclarationAndAssignment")
+private class ObservableCrossfade(
+    result: SuccessResult,
+    crossfadeDuration: Int,
+    clock: AnimationClockObservable
+) {
+    var painter: Painter by mutableStateOf(
+        createCrossfadePainter(
+            image = result.image,
+            saturation = 0f,
+            alpha = 0f,
+            brightness = 0f
+        )
+    )
+
+    var isFinished by mutableStateOf(false)
+        private set
+
+    private val animation: TransitionAnimation<CrossfadeTransition.State>
+
+    init {
+        animation = CrossfadeTransition.definition(crossfadeDuration).createAnimation(clock)
+
+        animation.onUpdate = {
+            // Animation tick, so update the painter using the current transition property values
+            painter = createCrossfadePainter(
+                image = result.image,
+                saturation = animation[CrossfadeTransition.Saturation],
+                alpha = animation[CrossfadeTransition.Alpha],
+                brightness = animation[CrossfadeTransition.Brightness]
+            )
+        }
+
+        animation.onStateChangeFinished = {
+            if (it == CrossfadeTransition.State.Loaded) {
+                isFinished = true
+            }
+        }
+    }
+
+    fun start() {
+        // Start the animation by transitioning to the Loaded state
+        animation.toState(CrossfadeTransition.State.Loaded)
+    }
+}
+
+private object CrossfadeTransition {
+    internal enum class State {
+        Loaded, Empty
+    }
+
+    internal val Alpha = FloatPropKey()
+    internal val Brightness = FloatPropKey()
+    internal val Saturation = FloatPropKey()
+
+    fun definition(durationMs: Int) = transitionDefinition {
+        state(State.Empty) {
+            this[Alpha] = 0f
```

Comment:
I wonder if these start values should be exposed (with defaults) and/or moved to constants.